### DISCUSSION TITLE
ctx.url.origin is always localhost:port / 0.0.0.0:port

### DISCUSSION THREAD

**Discussion by sharmapukar217** [rakkasjs/rakkasjs#76] [Q&A] [answered]
The origin is always localhost or 0.0.0.0 even opening from another device in the same Network. While developing i use both my pc and phone and since the origin is always localhost, the redirect from page guard is redirecting me to 0.0.0.0:3000/<url> on my phone's browser

**Comment by sharmapukar217** [comment]
One more thing to note.. ctx.url.origin is https://undefined when using with vite plugin mkcert 🤔

**Comment by cyco130** [comment]
This is a bug, thank you for reporting. I couldn't yet figure out what's wrong exactly but I will :)

**Reply by sharmapukar217** [reply]
> Comment options

After spending the whole night, I think there's something wrong with caching. The ctx ( or maybe ctx.url) is kind of cached so after opening on one browser, the url is same on other devices too

**Comment by cyco130** [comment]
It seems that `vite-plugin-mkcert` doesn't set the `Host` header, leaving us no way to determine the origin.

~~You can explicitly set the `HOST` env variable while I investigate it further. e.g. use: `HOST=192.168.0.16:4173 rakkas preview --https --host 0.0.0.0` (replace `192.168.0.16` with your local IP).~~

Sorry, ignore the second paragraph. It doesn't work because of a hattip bug.

**Comment by cyco130** [comment]
Sorry, updated the one up there. HatTip has a bug preventing it from working. Looking at it now.

**Comment by cyco130** [comment, marked as answer]
Fixed the HatTip bug. Upgrade `rakkasjs` to `0.6.2-next.9` and `@hattip/*` to `0.0.20` and use the `ORIGIN` env var like this: `ORIGIN=https://localhost:4173 rakkas preview --https` with the correct origin. I'll investigate why `vite-plugin-mkcert` doesn't set the `Host` header and why relative redirects don't pick up the correct origin.
